fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1938,4 +1947,58 @@ public SqlQueryData buildIncludeQuery(Class<?> resourceType, FHIRSearchContext s
         return query;
     }
 
+    /**
+     * Populates an IN clause with value set codes for a token search parameter specifying the
+     * :in or :not-in modifier.
+     *
+     * @param whereClauseSegment  - the segment to which the sub-segment will be added
+     * @param parameterValue      - the search parameter value - a ValueSet URL
+     * @param parameterTableAlias - the alias for the parameter table e.g. CPx
+     * @throws FHIRPersistenceException
+     */
+    private void populateValueSetCodesSubSegment(StringBuilder whereClauseSegment, String parameterValue,
+            String parameterTableAlias) throws FHIRPersistenceException {
+        final String METHODNAME = "populateValueSetCodesSubSegment";
+        log.entering(CLASSNAME, METHODNAME, parameterValue);
+
+        String tokenValuePredicateString = parameterTableAlias + DOT + TOKEN_VALUE + IN + LEFT_PAREN;
+        String codeSystemIdPredicateString = parameterTableAlias + DOT + CODE_SYSTEM_ID + IN + LEFT_PAREN;
+        String defaultCodeSystemId = nullCheck(identityCache.getCodeSystemId(JDBCConstants.DEFAULT_TOKEN_SYSTEM));
+        boolean codeSystemProcessed = false;
+
+        // Note: validation that the value set exists and is expandable was done when the
+        // search parameter was parsed, so does not need to be done here.
+        Map<String, Set<String>> codeSetMap = ValueSetSupport.getCodeSetMap(ValueSetSupport.getValueSet(parameterValue));
+        for (String codeSetUrl : codeSetMap.keySet()) {
+            Set<String> codes = codeSetMap.get(codeSetUrl);
+            if (codes != null) {
+                // Strip version from canonical codeSet URL. We don't store version in TOKEN_VALUES
+                // table so will just ignore it.
+                int index = codeSetUrl.lastIndexOf("|");
+                if (index != -1) {
+                    codeSetUrl = codeSetUrl.substring(0, index);
+                }
```

Comment:
*should* we store version in token_values? 🤔

Reply:
under what circumstances would we get a versioned url here?

Reply:
From what John told me, and looking at the code in ValueSetSupport.computeCodeSetMap() to verify, if a version is specified for the code system in the value set expansion, that version is appended. If no version is specified, a `|<version unknown>` is appended. So actually there should always be a version, meaning I probably don't need the check on line 1978, although I don't mind keeping it there as a fail-safe.